src/app/qgsdxfexportdialog.cpp

```diff
@@ -567,32 +579,22 @@ QgsDxfExport::SymbologyExport QgsDxfExportDialog::symbologyMode() const
 }
 
 
-void QgsDxfExportDialog::mFileSelectionButton_clicked()
-{
-  //get last dxf save directory
-  QgsSettings s;
-  QString lastSavePath = s.value( QStringLiteral( "qgis/lastDxfDir" ), QDir::homePath() ).toString();
-
-  QString filePath = QFileDialog::getSaveFileName( nullptr, tr( "Export as DXF" ), lastSavePath, tr( "DXF files *.dxf *.DXF" ) );
-  if ( !filePath.isEmpty() )
-  {
-    mFileLineEdit->setText( filePath );
-  }
-}
-
-
 void QgsDxfExportDialog::setOkEnabled()
 {
   QPushButton *btn = buttonBox->button( QDialogButtonBox::Ok );
 
-  QString filePath = mFileLineEdit->text();
+  QString filePath = mFileName->filePath();
   if ( filePath.isEmpty() )
   {
     btn->setEnabled( false );
+    return;
   }
 
   QFileInfo fi( filePath );
-  btn->setEnabled( fi.absoluteDir().exists() );
```

Comment:
Genuine question: why do we check the folder existence? If it doesn't exist yet, shouldn't we create it?